File: eCakeShop.Services/Database/Proizvod.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCakeShop.Services.Database
{
    public class Proizvod
    {
        public Proizvod()
        {
            NarudzbaProizvodis = new HashSet<NarudzbaProizvodi>();
        }
        public int ProizvodID { get; set; }
        public string? Naziv { get; set; }
        public string? Sifra { get; set; }
        public decimal Cijena { get; set; }
        public byte[]? Slika { get; set; }
        public string? Opis { get; set; }
        public int VrstaProizvodaID { get; set; }
        public virtual VrstaProizvoda VrstaProizvoda { get; set; } = null!;
        public virtual ICollection<NarudzbaProizvodi> NarudzbaProizvodis { get; set; }

    }
}

File: e_cakeshop/lib/models/proizvod.dart
import 'package:e_cakeshop/models/vrstaproizvoda.dart';
import 'package:json_annotation/json_annotation.dart';
part 'proizvod.g.dart';

@JsonSerializable()
class Proizvod {
  int? proizvodID;
  String? naziv;
  String? sifra;
  double? cijena;
  String? slika;
  String? opis;
  int? vrstaProizvodaID;
  VrstaProizvoda? vrstaProizvoda;

  Proizvod({
    this.proizvodID,
    this.naziv,
    this.sifra,
    this.cijena,
    this.slika,
    this.opis,
    this.vrstaProizvodaID,
    this.vrstaProizvoda,
  });

  factory Proizvod.fromJson(Map<String, dynamic> json) =>
      _$ProizvodFromJson(json);

  Map<String, dynamic> toJson() => _$ProizvodToJson(this);
}
